File: app/src/main/java/com/intacta/sosviagens/Utils/Utilities.java
package com.intacta.sosviagens.Utils;

import java.util.Calendar;

public class Utilities {
    public static final int RC_SIGN_IN = 123;
    public static boolean called = false;
    public static String callIdent="";
    public static String calltitle = "Números de emergência";
     public static boolean Isnight() {
        Calendar cal = Calendar.getInstance();
        int hour = cal.get(Calendar.HOUR_OF_DAY);
        return hour < 6 || hour > 18;
    }

}

File: app/src/main/java/com/intacta/sosviagens/Utils/Utilities.kt
package com.intacta.sosviagens.Utils

import android.app.Activity
import android.content.res.Configuration
import android.view.View
import androidx.core.view.ViewCompat
import io.reactivex.Completable
import io.reactivex.subjects.CompletableSubject


object Utilities {
    val RC_SIGN_IN = 123
    var calltitle = "Números de emergência"
    fun Isnight(activity: Activity): Boolean {

        return activity.resources.configuration.uiMode and
                Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
    }
    var REQUEST_CHECK_SETTINGS = 100


    fun fadeIn(view: View): Completable {
        val animationSubject = CompletableSubject.create()
        return animationSubject.doOnSubscribe {
            ViewCompat.animate(view)
                    .setDuration(1500)
                    .alpha(1f)
                    .withEndAction {
                        animationSubject.onComplete()
                    }
        }
    }
    fun fadeOut(view: View): Completable {
        val animationSubject = CompletableSubject.create()
        return animationSubject.doOnSubscribe {
            ViewCompat.animate(view)
                    .setDuration(1500)
                    .alpha(0f)
                    .withEndAction {
                        animationSubject.onComplete()
                        view.visibility = View.GONE
                    }
        }
    }
}
